SpeechResponder/ConfigurationWindow.xaml.cs

```diff
@@ -7,379 +7,404 @@
-            EDDI.Instance.SpeechResponderModalWait = false;
-            if ((bool)editScriptWindow.DialogResult)
-            {
-                updateScriptsConfiguration();
-                scriptsData.Items.Refresh();
-            }
-        }
+		private void eddiScriptsEnabledUpdated(object sender, RoutedEventArgs e)
+		{
+			if (sender is CheckBox checkbox)
+			{
+				if (checkbox.IsLoaded)
+				{
+					updateScriptsConfiguration();
+				}
+			}
+		}
 
-        private void viewScript(object sender, RoutedEventArgs e)
-        {
-            Script script = ((KeyValuePair<string, Script>)((Button)e.Source).DataContext).Value;
-            ViewScriptWindow viewScriptWindow = new ViewScriptWindow(script);
-            viewScriptWindow.Show();
-        }
+		private void eddiScriptsPriorityUpdated(object sender, SelectionChangedEventArgs e)
+		{
+			if (sender is ComboBox comboBox)
+			{
+				if (comboBox.IsLoaded && (comboBox.IsDropDownOpen || comboBox.IsKeyboardFocused))
+				{
+					updateScriptsConfiguration();
+				}
+			}
+		}
 
-        private void testScript(object sender, RoutedEventArgs e)
-        {
-            Script script = ((KeyValuePair<string, Script>)((Button)e.Source).DataContext).Value;
-            SpeechResponder responder = new SpeechResponder();
-            responder.Start();
-            // See if we have a sample
-            List<Event> sampleEvents;
-            object sample = Events.SampleByName(script.Name);
-            if (sample == null)
-            {
-                sampleEvents = new List<Event>();
-            }
-            else if (sample is string)
-            {
-                // It's as tring so a journal entry.  Parse it
-                sampleEvents = JournalMonitor.ParseJournalEntry((string)sample);
-            }
-            else if (sample is Event)
-            {
-                // It's a direct event
-                sampleEvents = new List<Event>() { (Event)sample };
-            }
-            else
-            {
-                Logging.Warn("Unknown sample type " + sample.GetType());
-                sampleEvents = new List<Event>();
-            }
+		private void editScript(object sender, RoutedEventArgs e)
+		{
+			Script script = ((KeyValuePair<string, Script>)((Button)e.Source).DataContext).Value;
+			OpenEditScriptWindow(script);
+		}
 
-            ScriptResolver scriptResolver = new ScriptResolver(Personality.Scripts);
-            if (sampleEvents.Count == 0)
-            {
-                sampleEvents.Add(null);
-            }
-            foreach (Event sampleEvent in sampleEvents)
-            {
-                responder.Say(scriptResolver, ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship monitor"))?.GetCurrentShip(), script.Name, sampleEvent, scriptResolver.priority(script.Name));
-            }
-        }
+		private void OpenEditScriptWindow(Script script)
+		{
+			EditScriptWindow editScriptWindow = new EditScriptWindow(Personality.Scripts, script.Name);
```

Comment:
The recovered script is not fully transferred to the script window window by this method.

Reply:
fixed

Reply:
Yes, that's better. :-)